cni/pkg/cmd/root.go

```diff
@@ -79,31 +79,46 @@ var rootCmd = &cobra.Command{
 			return
 		}
 
+		// Creates a basic health endpoint server that reports health status
+		// based on atomic flag, as set by installer
+		// TODO nodeagent watch server should affect this too, and drop atomic flag
+		installDaemonReady, watchServerReady := nodeagent.StartHealthServer()
+
 		if cfg.InstallConfig.AmbientEnabled {
 			// Start ambient controller
-			redirectMode := ambient.IptablesMode
-			if cfg.InstallConfig.EbpfEnabled {
-				redirectMode = ambient.EbpfMode
-			}
-			server, err := ambient.NewServer(ctx, ambient.AmbientArgs{
-				SystemNamespace: ambient.PodNamespace,
-				Revision:        ambient.Revision,
-				RedirectMode:    redirectMode,
-				LogLevel:        cfg.InstallConfig.LogLevel,
-			})
+			redirectMode := nodeagent.InPodMode
+
+			// node agent will spawn a goroutine and watch the K8S API for events,
+			// as well as listen for messages from the CNI binary.
+			log.Infof("Starting ambient node agent with redirect mode %s", redirectMode)
+			ambientAgent, err := nodeagent.NewServer(ctx, watchServerReady, cfg.InstallConfig.CNIEventAddress,
+				nodeagent.AmbientArgs{
+					SystemNamespace: nodeagent.PodNamespace,
+					Revision:        nodeagent.Revision,
+					RedirectMode:    redirectMode,
```

Comment:
Agree, yes. We had a mode flag in the previous impl, because we had multiple modes.


In this impl this is the only mode, so I can remove this as strictly useless (it is), but I am also fine to leave it in for future work.

Reply:
I think there's an argument to be made that if you were implementing an alt mode, having this flag in place signals where in the code we are doing "inpod-specific" stuff and might be useful in that way.

But I cannot really argue with the fact that it does nothing in this PR, so I'll remove it.